test/libsolidity/syntaxTests/using/library_functions_attached_in_single_directive_inside_contract.sol

```diff
@@ -0,0 +1,17 @@
+library L {
+    function publicFunction(uint) public pure {}
+    function externalFunction(uint) external pure {}
+    function internalFunction(uint) internal pure {}
+}
+
+contract C {
+    using {L.publicFunction, L.externalFunction, L.internalFunction} for uint;
```

Comment:
Actually, there's one more test you should add.

`using {publicFunction, externalFunction, internalFunction} for uint;` in a library. I.e. inside a library you can use functions in `using for` without qualifying them with `L.`. Please make sure that this now works with external functions.

Reply:
I added the test and external functions are allowed in `using for` inside the library. But the call to the external function itself, as well as the to the public one, are not supposed to be valid, right?

Reply:
Good question.

The public one is definitely valid because you can call public functions internally as `f()` inside a library.

As for external, I think it should still work and make an external call because it's not the `using for` that decides whether the call is external or not but how and where you call it. Now you can call an external function via `using for` in a contract so I think it should work in a library too. The way it's specified in `using for` should not make any difference.

Reply:
Ok, thanks for the explanation, cameel. I will work on the necessary changes.

Reply:
Actually, it seems this is was already happening in `0.8.17`. 

```solidity
library L {
    function publicFunction(uint) public pure {}

    using {publicFunction} for uint;

    function f() public pure {
        uint x;
        x.publicFunction();
    }
}

// Error: Libraries cannot call their own functions externally.
```

Reply:
Ah, true. It's not allowed in libraries. Forgot about that.

Still, what I meant is that this should behave the same way no matter whether you use `L.f` or `f` in `using for`. If it's not allowed, both should produce the error you're showing. We should not accept one and reject the other in `using for`.